File: backend/HSE_Football_Backend/Models/TeamApplication.cs
using System.ComponentModel.DataAnnotations;

namespace HSE_Football_Backend.Models
{
	/// <summary>
	/// Заявка команды
	/// </summary>
	public class TeamApplication
	{
		/// <summary>
		/// ID заявки команды
		/// </summary>
		[Key]
		public long Id { get; set; }

		/// <summary>
		/// ID команды
		/// </summary>
		public long TeamId { get; set; }

        /// <summary>
        /// Название 
        /// </summary>
        [StringLength(41, MinimumLength = 2, ErrorMessage = "Название должно быть от 2 до 40 символов")]
        [RegularExpression(@"[А-Яа-яЁёA-Za-z]+$", ErrorMessage = "В названии могут присутствовать только буквы")]
        public string? Name { get; set; }

        /// <summary>
        /// Логотип
        /// </summary>
        public string? Logo { get; set; }

        /// <summary>
        /// Контактная информация
        /// </summary>
        [StringLength(41, MinimumLength = 2, ErrorMessage = "Контактная информация должна быть от 2 до 40 символов")]
        public string? Contact { get; set; }

        /// <summary>
        /// Игровое амплуа
        /// Ниже представлены степени двойки, то есть значение для ЦФРВ - 2^0 = 1
        /// 0 - ЦФРВ
        /// 1 - ЛФРВ
        /// 2 - ПФРВ
        /// 3 - ЦАП
        /// 4 - ЦОП
        /// 5 - ЛП
        /// 6 - ПП
        /// 7 - ЦЗ
        /// 8 - ЛЗ
        /// 9 - ПЗ
        /// 10 - ВРТ
        /// </summary>
        [Range(0, 2047, ErrorMessage = "Некорректное амплуа")]
		public int PlayerPosition { get; set; }

		/// <summary>
		/// Турниры, на которые ищутся игроки
		/// Ниже представлены степени двойки, то есть значение для Второй и Первой лиг - 2^6 + 2^7 = 192
		/// 0 - Осенний кубок
		/// 1 - Футзал Молодежная лига
		/// 2 - Футзал Вторая лига
		/// 3 - Футзал Первая лига
		/// 4 - Футзал Высшая лига
		/// 5 - Молодежная лига
		/// 6 - Вторая лига
		/// 7 - Первая лига
		/// 8 - Высшая лига
		/// 9 - Летний кубок
		/// </summary>
		[Range(0, 1023, ErrorMessage = "Некорректные турниры")]
		public int Tournaments { get; set; }

		/// <summary>
		/// Описание
		/// </summary>
		[StringLength(201, ErrorMessage = "Слишком много символов")]
		public string? Description { get; set; }

		/// <summary>
		/// Показывает возможное наличие нецензурной лексики
		/// </summary>
		public bool Attention { get; set; }
	}
}

File: android/sources/features/team/domain_team/src/main/kotlin/andryuh/football/domain_team/dto/TeamApplication.kt
package andryuh.football.domain_team.dto

import android.os.Parcelable
import androidx.compose.runtime.Immutable
import andryuh.football.core_models.PlayerPosition
import andryuh.football.core_models.Tournament
import andryuh.football.core_network.serializers.EnumListAsIntSerializer
import kotlinx.parcelize.Parcelize
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Immutable
@Parcelize
@Serializable
data class TeamApplication(
  val id: String?,
  val teamId: String,
  val name: String,
  @SerialName("logo") val imageUrl: String?,
  val description: String?,
  val contact: String?,
  @Serializable(with = PlayerPositionsSerializer::class) val playerPosition: List<PlayerPosition>,
  @Serializable(with = TournamentsSerializer::class) val tournaments: List<Tournament>,
) : Parcelable

@Immutable
@Parcelize
@Serializable
data class TeamApplicationCreation(
  val teamId: String,
  val name: String,
  @SerialName("logo") val imageUrl: String?,
  val description: String?,
  val contact: String?,
  @Serializable(with = PlayerPositionsSerializer::class) val playerPosition: List<PlayerPosition>,
  @Serializable(with = TournamentsSerializer::class) val tournaments: List<Tournament>,
) : Parcelable

internal class PlayerPositionsSerializer :
  EnumListAsIntSerializer<PlayerPosition>(
    serialName = "playerPosition",
    enumArray = PlayerPosition.values(),
  )

internal class TournamentsSerializer :
  EnumListAsIntSerializer<Tournament>(
    serialName = "tournaments",
    enumArray = Tournament.values(),
  )
